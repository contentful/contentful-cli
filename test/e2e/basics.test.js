import { join } from 'path'
import { platform } from 'os'

import execa from 'execa'

let cmd = null
switch (platform()) {
  case 'win32':
    cmd = 'contentful-cli-win.exe'
    break
  case 'linux':
    cmd = 'contentful-cli-linux'
    break
  case 'darwin':
    cmd = 'contentful-cli-macos'
    break
}
if (!cmd) {
  throw new Error(`Platform ${platform()} is not supported`)
}

const packageVersion = require('../../package.json').version
const cwd = join(__dirname, '../../', 'build')
cmd = join(cwd, cmd)

test('should return code 1 when errors exist no args', async () => {
  try {
    await execa.shell(cmd)
    throw new Error('CLI should exit with error')
  } catch (error) {
    expect(error.code).toBe(1)
    expect(error.stderr).toMatchSnapshot()
  }
})

test('should print help message', async () => {
  const {stdout} = await execa.shell(`${cmd} --help`)
  expect(stdout).toMatchSnapshot()
})

test('should print help message on shortcut', async () => {
  const {stdout} = await execa.shell(`${cmd} -h`)
  expect(stdout).toMatchSnapshot()
})

test('should print help message on wrong subcommand', async () => {
  try {
    await execa.shell(`${cmd} lolbar`)
    throw new Error('CLI should exit with error')
  } catch (error) {
    expect(error.code).toBe(1)
    expect(error.stderr).toMatchSnapshot()
  }
})

test('should print version number', async () => {
  const {stdout} = await execa.shell(`${cmd} --version`)
  await expect(stdout).toContain(packageVersion)
})
