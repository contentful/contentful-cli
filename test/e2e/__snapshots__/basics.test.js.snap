// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should print help message 1`] = `
"
Usage: contentful <cmd> [args]

Commands:
   completion    generate bash completion script
   boilerplate   Download a boilerplate
   config        Manage and list your configs
   content-type  Manage and list your space content types
                                                                   [aliases: ct]
   extension     Manage and list your extensions
   guide         A guide introducing basic concepts of working with
                              Contentful
   login         Login to Contentful
   logout        Logout from Contentful
   space         Manage and list your spaces

Options:
  -h, --help     Show help                                             [boolean]
  -v, --version  Show current version                                  [boolean]

Copyright 2018 Contentful, this is a BETA release"
`;

exports[`should print help message on shortcut 1`] = `
"
Usage: contentful <cmd> [args]

Commands:
   completion    generate bash completion script
   boilerplate   Download a boilerplate
   config        Manage and list your configs
   content-type  Manage and list your space content types
                                                                   [aliases: ct]
   extension     Manage and list your extensions
   guide         A guide introducing basic concepts of working with
                              Contentful
   login         Login to Contentful
   logout        Logout from Contentful
   space         Manage and list your spaces

Options:
  -h, --help     Show help                                             [boolean]
  -v, --version  Show current version                                  [boolean]

Copyright 2018 Contentful, this is a BETA release"
`;

exports[`should print help message on wrong subcommand 1`] = `
"
Usage: contentful <cmd> [args]

Commands:
   completion    generate bash completion script
   boilerplate   Download a boilerplate
   config        Manage and list your configs
   content-type  Manage and list your space content types
                                                                   [aliases: ct]
   extension     Manage and list your extensions
   guide         A guide introducing basic concepts of working with
                              Contentful
   login         Login to Contentful
   logout        Logout from Contentful
   space         Manage and list your spaces

Options:
  -h, --help     Show help                                             [boolean]
  -v, --version  Show current version                                  [boolean]

Copyright 2018 Contentful, this is a BETA release
Please specify a command.
"
`;

exports[`should return code 1 when errors exist no args 1`] = `
"
Usage: contentful <cmd> [args]

Commands:
   completion    generate bash completion script
   boilerplate   Download a boilerplate
   config        Manage and list your configs
   content-type  Manage and list your space content types
                                                                   [aliases: ct]
   extension     Manage and list your extensions
   guide         A guide introducing basic concepts of working with
                              Contentful
   login         Login to Contentful
   logout        Logout from Contentful
   space         Manage and list your spaces

Options:
  -h, --help     Show help                                             [boolean]
  -v, --version  Show current version                                  [boolean]

Copyright 2018 Contentful, this is a BETA release
Please specify a command.
"
`;
